system/jlib/jfile.cpp

```diff
@@ -1827,10 +1827,13 @@ void CFileIO::close()
 {
     if (file != NULLFILE)
     {
-        if (!CloseHandle(file))
+        // ensure file handle is cleared before throwing exception, to avoid dtor recalling close() as stack unwinds.
+        HANDLE tmpHandle = NULLFILE;
```

Comment:
personal: I would have probably coded as 

```
HANDLE tmpHandle = file;
file = NULLFILE;
```

rather than using swap, but I suspect the generated code will be identical.